src/neo/Wallets/Wallet.cs

```diff
@@ -565,11 +565,13 @@ public long CalculateNetworkFee(DataCache snapshot, Transaction tx)
                     var contract = NativeContract.ContractManagement.GetContract(snapshot, hash);
                     if (contract is null)
                         throw new ArgumentException($"The smart contract or address {hash} is not found");
-                    var md = contract.Manifest.Abi.GetMethod("verify", 0);
+                    var md = contract.Manifest.Abi.GetMethod("verify", -1);
                     if (md is null)
-                        throw new ArgumentException($"The smart contract {contract.Hash} haven't got verify method without arguments");
+                        throw new ArgumentException($"The smart contract {contract.Hash} haven't got verify method");
                     if (md.ReturnType != ContractParameterType.Boolean)
                         throw new ArgumentException("The verify method doesn't return boolean value.");
+                    if (md.Parameters.Length > 0 && invocationScript is null)
+                        throw new ArgumentException("The verify method needs invocation script.");
```

Comment:
Maybe you want to keep error messages short on purpose, but I think it would help decrease devs debugging time if the message was a bit more elaborate, like:
"The verify method requires parameters that need to be passed via the wintess' invocation script."